source/file_layout.rst

```diff
@@ -169,24 +169,39 @@ Each block begins with the following header:
   bytes.
 
 - ``used_size`` (64-bit unsigned integer, big-endian): The amount of
-  used space for the block (not including the header), in bytes.
+  used space for the block on disk (not including the header), in
+  bytes.
+
+- ``memory_size`` (64-bit unsigned integer, big-endian): The size of
+  the block when decoded, in bytes.  If ``encoding`` is all zeros
+  (indicating no encoding), it **must** be equal to ``used_size``.  If
+  there is an encoding, this is the size of the decoded block data,
+  which **may** be different from the size of the encoded block in the
+  ASDF file.
```

Comment:
I'm going to nitpick over what to call this.  The fact that this involves decompressing "in memory" I think is an implementation detail (for example, you could decompress directly to disk or something).  Maybe `actual_size` (if only because it reference a They Might Be Giants song)?